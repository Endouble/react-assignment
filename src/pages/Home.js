import React from 'react';

import Layout from '../modules/Layout/Layout.component';

const App = () => (
    <Layout>
        <header>
            <h1>Welcome to React</h1>
        </header>
        <p>
            To get started, edit <code>src/App.js</code> and save to reload.
        </p>
    </Layout>
);

export default App;
