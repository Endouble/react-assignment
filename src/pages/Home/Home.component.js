import React from 'react';
import PropTypes from 'prop-types';

import { childrenWithProps } from '../../utils';

import Header from '../../modules/Header';
import Footer from '../../modules/Footer';
import PokeDex from '../../modules/PokeDex';

export const CardsContext = React.createContext([]);
export const FilterContext = React.createContext();

class ContextProvider extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            cards: [],
            filter: {
                value: '',
                filterBy: 'name',
            },
        };
        this.setCards = this.setCards.bind(this);
        this.setFilter = this.setFilter.bind(this);
    }
    setCards(cards) {
        this.setState({
            cards,
        });
    }
    setFilter(filter) {
        this.setState({
            filter,
        });
    }

    render() {
        const children = childrenWithProps(this.props.children, { setCards: this.setCards, setFilter: this.setFilter });
        return (
            <CardsContext.Provider value={this.state.cards}>
                <FilterContext.Provider value={this.state.filter}>
                    {children}
                </FilterContext.Provider>
            </CardsContext.Provider>
        );
    }
}

ContextProvider.propTypes = {
    children: PropTypes.node,
};

ContextProvider.defaultProps = {
    children: null,
};

const Home = () => (
    <ContextProvider>
        <Header />
        <PokeDex />
        <Footer />
    </ContextProvider>
);

export default Home;
