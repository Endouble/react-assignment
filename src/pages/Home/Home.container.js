import React from 'react';
import PropTypes from 'prop-types';

import Header from '../../modules/Header';
import Footer from '../../modules/Footer';
import PokeDex from '../../modules/PokeDex';

import { CardsContext, FilterContext, IsLoadingContext,
    defaultFilter, defaultCards, defaultIsLoading } from '../../context';

class Home extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            cards: defaultCards,
            filter: defaultFilter,
            isLoading: defaultIsLoading,
        };
        this.setCards = this.setCards.bind(this);
        this.setFilter = this.setFilter.bind(this);
        this.toggleIsLoading = this.toggleIsLoading.bind(this);
    }
    setCards(cards) {
        this.setState({
            cards,
        });
    }
    setFilter(filter) {
        this.setState({
            filter,
        });
    }
    toggleIsLoading(isLoading) {
        this.setState({
            isLoading,
        });
    }

    render() {
        const childrenProps = {
            setCards: this.setCards,
            setFilter: this.setFilter,
            toggleIsLoading: this.toggleIsLoading,
        };
        return (
            <CardsContext.Provider value={this.state.cards}>
                <FilterContext.Provider value={this.state.filter}>
                    <IsLoadingContext.Provider value={this.state.isLoading}>
                        <Header />
                        <PokeDex {...childrenProps} />
                        <Footer setFilter={this.setFilter} />
                    </IsLoadingContext.Provider>
                </FilterContext.Provider>
            </CardsContext.Provider>
        );
    }
}

Home.propTypes = {
    children: PropTypes.node,
};

Home.defaultProps = {
    children: null,
};

export default Home;
