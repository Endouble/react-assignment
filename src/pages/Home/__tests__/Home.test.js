import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from '../';

configure({ adapter: new Adapter() });

describe('Home', () => {
    it('renders correctly', () => {
        const component = shallow(<Home />);
        expect(component).toMatchSnapshot();
    });
});
