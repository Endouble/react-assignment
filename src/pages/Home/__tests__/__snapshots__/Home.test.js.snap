// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<ContextProvider
  value={Array []}
>
  <ContextProvider
    value={
      Object {
        "filterBy": "name",
        "value": "",
      }
    }
  >
    <ContextProvider
      value={false}
    >
      <Header />
      <PokeDexData
        setCards={[Function]}
        setFilter={[Function]}
        toggleIsLoading={[Function]}
      />
      <FooterData
        setFilter={[Function]}
      />
    </ContextProvider>
  </ContextProvider>
</ContextProvider>
`;
